greenboot.spec

```diff
@@ -41,99 +41,122 @@ Requires:           rpm-ostree
+Provides: bundled(crate(pest)) = 2.5.6
+Provides: bundled(crate(pest_derive)) = 2.5.6
+Provides: bundled(crate(pest_generator)) = 2.5.6
+Provides: bundled(crate(pest_meta)) = 2.5.6
 Provides: bundled(crate(pin-utils)) = 0.1.0
 Provides: bundled(crate(pretty_env_logger)) = 0.4.0
-Provides: bundled(crate(proc-macro-error)) = 1.0.4
-Provides: bundled(crate(proc-macro-error-attr)) = 1.0.4
-Provides: bundled(crate(proc-macro2)) = 1.0.43
+Provides: bundled(crate(proc-macro2)) = 1.0.54
 Provides: bundled(crate(quick-error)) = 1.2.3
-Provides: bundled(crate(quote)) = 1.0.21
-Provides: bundled(crate(regex)) = 1.6.0
-Provides: bundled(crate(regex-syntax)) = 0.6.27
+Provides: bundled(crate(quote)) = 1.0.26
+Provides: bundled(crate(regex)) = 1.7.3
+Provides: bundled(crate(regex-syntax)) = 0.6.29
 Provides: bundled(crate(ron)) = 0.7.1
 Provides: bundled(crate(rust-ini)) = 0.18.0
-Provides: bundled(crate(ryu)) = 1.0.11
-Provides: bundled(crate(serde)) = 1.0.144
-Provides: bundled(crate(serde_derive)) = 1.0.144
-Provides: bundled(crate(serde_json)) = 1.0.85
-Provides: bundled(crate(sha1)) = 0.10.5
+Provides: bundled(crate(rustix)) = 0.37.4
+Provides: bundled(crate(ryu)) = 1.0.13
+Provides: bundled(crate(serde)) = 1.0.159
+Provides: bundled(crate(serde_derive)) = 1.0.159
+Provides: bundled(crate(serde_json)) = 1.0.95
+Provides: bundled(crate(sha2)) = 0.10.6
 Provides: bundled(crate(strsim)) = 0.10.0
-Provides: bundled(crate(syn)) = 1.0.100
-Provides: bundled(crate(termcolor)) = 1.1.3
-Provides: bundled(crate(thiserror)) = 1.0.35
-Provides: bundled(crate(thiserror-impl)) = 1.0.35
-Provides: bundled(crate(toml)) = 0.5.9
-Provides: bundled(crate(typenum)) = 1.15.0
+Provides: bundled(crate(syn)) = 1.0.109
+Provides: bundled(crate(syn)) = 2.0.11
+Provides: bundled(crate(termcolor)) = 1.2.0
+Provides: bundled(crate(thiserror)) = 1.0.40
+Provides: bundled(crate(thiserror-impl)) = 1.0.40
+Provides: bundled(crate(toml)) = 0.5.11
+Provides: bundled(crate(typenum)) = 1.16.0
 Provides: bundled(crate(ucd-trie)) = 0.1.5
-Provides: bundled(crate(unicode-ident)) = 1.0.4
+Provides: bundled(crate(unicode-ident)) = 1.0.8
+Provides: bundled(crate(utf8parse)) = 0.2.1
 Provides: bundled(crate(version_check)) = 0.9.4
 Provides: bundled(crate(wasi)) = 0.11.0+wasi_snapshot_preview1
 Provides: bundled(crate(winapi)) = 0.3.9
 Provides: bundled(crate(winapi-i686-pc-windows-gnu)) = 0.4.0
 Provides: bundled(crate(winapi-util)) = 0.1.5
 Provides: bundled(crate(winapi-x86_64-pc-windows-gnu)) = 0.4.0
+Provides: bundled(crate(windows-sys)) = 0.45.0
+Provides: bundled(crate(windows-targets)) = 0.42.2
+Provides: bundled(crate(windows_aarch64_gnullvm)) = 0.42.2
+Provides: bundled(crate(windows_aarch64_msvc)) = 0.42.2
+Provides: bundled(crate(windows_i686_gnu)) = 0.42.2
+Provides: bundled(crate(windows_i686_msvc)) = 0.42.2
+Provides: bundled(crate(windows_x86_64_gnu)) = 0.42.2
+Provides: bundled(crate(windows_x86_64_gnullvm)) = 0.42.2
+Provides: bundled(crate(windows_x86_64_msvc)) = 0.42.2
```

Comment:
It seems odd that we need to include all these Windows related crates as part of packaging

Can we adjust the `Cargo.toml` in a way where we don't have to include Windows crates?

Reply:
Yeah...I did my own investigation into removing these windows-specific crates and there doesn't appear to be good support for indicating that your Rust project only supports Linux

We could vendor the dependencies and use https://github.com/coreos/cargo-vendor-filterer to be more exclusive, but that would be an entirely different conversation.

Reply:
in FDO we just rm -rf them from the vendor directory that we use to build fdo in rhel (maybe fedora too @nullr0ute ) - we could do the same here and just leave off these bundled things

Reply:
In the coreos/ and containers/ orgs we use cargo-vendor-filterer and I've also switched our Fedora packages to use vendored dependencies consistently too.